### ISSUE TITLE
Don't skip training test

### ISSUE THREAD

**Issue by constantinpape** [computational-cell-analytics/micro-sam#751]
The training test is very slow since yesterday. I am skipping it for now, see:
https://github.com/computational-cell-analytics/micro-sam/blob/master/test/test_training.py#L13-L14

But we should figure out why it's slower and reactivate it so that we notice if some change affects training.

I am not sure what the reason is for this slowdown. Nothing has changed in the training code, it seems to happen irrespective of the pytorch version, so is also not just for pytorch 2.5, and everything still works normally for me locally.

**Comment by DavidMed13** [comment]
Hi, Im David Medina, from México, im trying to train my own model but de training is not moving forward, I don't know if this commend is related to this issue im having, by the way, great work with usam 

<img width="745" alt="Captura de pantalla 2024-10-24 a la(s) 10 10 54 p m" src="https://github.com/user-attachments/assets/534e6f4c-78db-4bec-a012-5614e8920095">

**Comment by DavidMed13** [comment]
Sure! 
<img width="460" alt="Captura de pantalla 2024-10-25 a la(s) 1 35 06 p m" src="https://github.com/user-attachments/assets/6fae9801-b581-4196-9866-d1dde375b746">

**Comment by constantinpape** [comment]
Hi @DavidMed13 , your issue is likely due to the fact that training on the MPS devices is very slow. You can try to train on the CPU by setting `device="cpu"`. But please note that training on the CPU will still be quite slow and you will likely have to wait several hours and also need sufficient main memory (ideally >=32 GB).

This is why we recommend using a GPU for training. If you don't have access to a GPU you can use cloud resources for this, see https://github.com/computational-cell-analytics/micro-sam/blob/master/notebooks/sam_finetuning.ipynb for details.

**Comment by DavidMed13** [comment]
Hello, thank you for the information, I have tried put it in cpu but still won't advance, I don't know if im doing something wrong
 
If I use the same coding form the notebook I get this error
<img width="1323" alt="Captura de pantalla 2024-11-08 a la(s) 1 07 35 p m" src="https://github.com/user-attachments/assets/a290d05c-1d04-475a-a9e4-8b1cfb477a25">

But if I delete "rois" work but I don't know if that modification is the problem, because I guess it shouldn´t take so much time is a little training is only 5 imagines with labels
<img width="1271" alt="Captura de pantalla 2024-11-08 a la(s) 1 06 13 p m" src="https://github.com/user-attachments/assets/49efc86c-dd04-475e-b7ad-587a11b9bd89">


I have tried to replicate this in google collab and kaggle but I always have error importing the packages 

I really want to create this fine-tuning model because is for microglia, I really appreciate if you can help me, I really feel like I have tried all haha 

Thank you in advance :D

**Comment by DavidMed13** [comment]
Sure :)
<img width="356" alt="Captura de pantalla 2024-11-08 a la(s) 2 49 38 p m" src="https://github.com/user-attachments/assets/37066720-2710-4d93-8bcc-5558185ba350">

**Comment by anwai98** [comment]
Okay, looks like you are using the latest `micro-sam` already. Hmm, that's strange.

I'll try to reproduce the issue you mentioned above.

Meanwhile, could you confirm two things for us:
- The installation process (via conda-forge / source)?
- Are you making use of the latest finetuning notebook? (https://github.com/computational-cell-analytics/micro-sam/blob/master/notebooks/sam_finetuning.ipynb)

**Comment by DavidMed13** [comment]
The installation was via mamba in a environment and yes im using the last fine-tuning notebook :)

**Comment by anwai98** [comment]
Hi @DavidMed13,

Thanks for sharing the details.

Another request: could you run this and send me the outputs?

```bash
python -c "import inspect; from micro_sam.training.training import default_sam_loader; print(inspect.getsource(default_sam_loader))"
```

**Comment by DavidMed13** [comment]
Of course

<img width="754" alt="Captura de pantalla 2024-11-08 a la(s) 3 19 12 p m" src="https://github.com/user-attachments/assets/f6b58aac-c649-4068-8ab3-1e06ee82af24">

And thank you a lot @anwai98

**Comment by anwai98** [comment]
Ah yeah, I see the issue now. Seems like we recently updated this part.

Can I request you to install `micro-sam` from source and try again? (see suggestion below)

Since you already have `micro-sam` installed, it's a rather easier pathway to installing the package from source:
- Clone our repo
- Enter the repo and install `micro-sam` in development mode
```bash
mamba activate <INSTALLED_ENVIRONMENT_NAME>  # where, you should ensure that you activate the environment where micro-sam is already installed
git clone https://github.com/computational-cell-analytics/micro-sam
cd micro-sam
pip install -e .
```

**Comment by DavidMed13** [comment]
Okay, done I did it, should I try again the notebook?

<img width="567" alt="Captura de pantalla 2024-11-08 a la(s) 3 31 15 p m" src="https://github.com/user-attachments/assets/9ceb9575-a47b-487d-97e1-759595a0c3f0">

**Comment by anwai98** [comment]
Yes, let's try training the model again and see if the error is (hopefully) gone!

**Comment by DavidMed13** [comment]
Okay it work! 
Now i don't know how much time should I wait if im in my MacBook,
Like 4 hrs? 
Yes I used the the rois hehe :)
<img width="475" alt="Captura de pantalla 2024-11-08 a la(s) 3 39 08 p m" src="https://github.com/user-attachments/assets/5d72dbf2-6576-4319-ac53-a9d3bab38d40">

And now im in here
<img width="887" alt="Captura de pantalla 2024-11-08 a la(s) 3 39 42 p m" src="https://github.com/user-attachments/assets/066e3968-127f-47c4-8026-f6ac783dad17">

**Comment by anwai98** [comment]
Okay, that's great to see that the error is gone.

Regarding the training runtime, you should see the progress bar going forward (i.e. processing a few iterations) in a couple of minutes (and the overall training would probably take a couple of hours). If it's slower than what you would desire, I can suggest a) using Kaggle (as it provides reasonable GPU resources free to use) or b) in case you have access to some compute cluster resources. This would free up the training overhead on your laptop.

**Comment by DavidMed13** [comment]
Hey thank you very much for all the help, I will try it on kaggle, my lab
have a server I will ask for permission

Thank you a lot :)
